### ISSUE TITLE
CMake configure failure on CentOS 7 Intel system

### ISSUE THREAD

**Issue by scivision** [zlib-ng/zlib-ng#957]
Zlib 2.0.2
Intel x86_64 with CentOS 7

```sh
cmake -B build -DZLIB_COMPAT=on
```

results in

```
-- Using CMake version 3.20.0
-- ZLIB_HEADER_VERSION: 1.2.11
-- ZLIBNG_HEADER_VERSION: 2.0.2
-- Arch detected: 'x86_64'
-- Basearch of 'x86_64' has been detected as: 'x86'
-- Architecture supports unaligned reads
-- Architecture supports unaligned reads of > 4 bytes
-- sizeof(void *) is  bytes
CMake Error at CMakeLists.txt:450 (message):
  sizeof(void *) is neither 32 nor 64 bit
```

A user on a system I don't have access to emailed me this. I can ask them for more information, as CentOS 7 on my own computer doesn't have this problem. Zlib 1.2.11 does work for them.

Would you like their CMakeError.log?

**Comment by nmoinvaz** [comment]
Yes we need that file to figure out where it is failing. Thanks.

**Comment by scivision** [comment]
[CMakeError.log](https://github.com/zlib-ng/zlib-ng/files/6469430/CMakeError.log)

**Comment by nmoinvaz** [comment]
Appears to be the same as #956 & #959.

**Comment by scivision** [comment]
appears to be fixed by #959
